lib/nrf_modem_lib/sanity.c

```diff
@@ -91,3 +98,96 @@ BUILD_ASSERT(ESHUTDOWN       == NRF_ESHUTDOWN,       "Errno not aligned with nrf
+
+BUILD_ASSERT(SOCK_STREAM == NRF_SOCK_STREAM, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(SOCK_DGRAM == NRF_SOCK_DGRAM, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(SOCK_RAW == NRF_SOCK_RAW, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(IPPROTO_IP == NRF_IPPROTO_IP, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_TCP == NRF_IPPROTO_TCP, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_UDP == NRF_IPPROTO_UDP, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_IPV6 == NRF_IPPROTO_IPV6, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_RAW == NRF_IPPROTO_RAW, "Socket value not aligned with modemlib.");
+/* Comparison is inverted here to prevent `WARNING:CONSTANT_COMPARISON` from checkpatch */
+BUILD_ASSERT(NRF_SPROTO_TLS1v2 == IPPROTO_TLS_1_2, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(NRF_SPROTO_DTLS1v2 == IPPROTO_DTLS_1_2, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(SOL_TLS == NRF_SOL_SECURE, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(SOL_SOCKET == NRF_SOL_SOCKET, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(MSG_DONTWAIT == NRF_MSG_DONTWAIT, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(MSG_PEEK == NRF_MSG_PEEK, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(MSG_WAITALL == NRF_MSG_WAITALL, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(AI_CANONNAME == NRF_AI_CANONNAME, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(AI_NUMERICSERV == NRF_AI_NUMERICSERV, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(AI_PDNSERV == NRF_AI_PDNSERV, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(POLLIN == NRF_POLLIN, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(POLLOUT == NRF_POLLOUT, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(POLLERR == NRF_POLLERR, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(POLLHUP == NRF_POLLHUP, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(POLLNVAL == NRF_POLLNVAL, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(TLS_SEC_TAG_LIST == NRF_SO_SEC_TAG_LIST, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(TLS_HOSTNAME == NRF_SO_SEC_HOSTNAME, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(TLS_CIPHERSUITE_LIST == NRF_SO_SEC_CIPHERSUITE_LIST,
+	     "Socket value not aligned with modemlib.");
+BUILD_ASSERT(TLS_PEER_VERIFY == NRF_SO_SEC_PEER_VERIFY, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(TLS_DTLS_ROLE == NRF_SO_SEC_ROLE, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(TLS_SESSION_CACHE == NRF_SO_SEC_SESSION_CACHE,
+	     "Socket value not aligned with modemlib.");
+BUILD_ASSERT(TLS_SESSION_CACHE_PURGE == NRF_SO_SEC_SESSION_CACHE_PURGE,
+	     "Socket value not aligned with modemlib.");
+BUILD_ASSERT(TLS_DTLS_HANDSHAKE_TIMEO == NRF_SO_SEC_DTLS_HANDSHAKE_TIMEO,
```

Comment:
I'm not 100% certain it's the right move (specifically for TLS options, which are not covered by any standard). Currently, the "downstream" options (i.e. those not implemented by Zephyr, but provided only for compability with modem lib) start just after the "core" Zephyr TLS options. Now, in case some community member decides to add a new TLS option to Zephyr (which already happened several times in the past), we'd need to bump the "downstream" option number as well. This means, we'd need a new modem lib release just to deal with that.

Reply:
@rlubos That's a good point. Are there any other cases like TLS values that you know of?

Reply:
TLS socket options were my major concern. Regular socket options are also a bit debatable, but we seem to specify modem-specific option numbers at certain offset, so the risk here is mitigated. Other symbols AFAICT shouldn't change.

Reply:
@rlubos Decided to re-introduce translation for all of the socket options since they are in general the most likely to not follow a standard.